packages/puppeteer-core/src/cdp/Page.ts

```diff
@@ -257,12 +257,12 @@ export class CdpPage extends Page {
   }
 
   async #onActivation(newSession: CDPSession): Promise<void> {
-    this.#primaryTargetClient = newSession;
     assert(
-      this.#primaryTargetClient instanceof CdpCDPSession,
-      'CDPSession is not instance of CDPSessionImpl',
+      newSession instanceof CdpCDPSession,
```

Comment:
In practice we don't need this check as the target emits a CdpCDPSession only, but that would require another refactor to events. Can we just add a comment.
```js
// TODO: Remove once we have separate Event type for CdpCDPSession.
```
So that it can be fixed separate.

Reply:
Done